doc/man3/flux_kvs_lookup.rst

```diff
@@ -152,9 +152,9 @@ FLUX_KVS_WATCH_APPEND
    Specified along with FLUX_KVS_WATCH, this flag will alter watch
    behavior to only respond when :var:`key` is mentioned verbatim in a
    committed transaction and the key has been appended to. The response
-   will only contain the additional appended data. Note that only data
-   length is considered for appends and no guarantee is made that prior
-   data hasn't been overwritten.
+   will only contain the additional appended data.  This flag will
+   only work with appends with the FLUX_KVS_APPEND.  Overwritten data
+   will result in an error.
```

Comment:
Since the only way to append is with FLUX_KVS_APPEND,  I think the first sentence added here could be dropped.

The second one might be slightly clearer if it read
> If the value is overwritten, the lookup fails with EINVAL

(or whatever errno it fails wiith)

Reply:
Aside: if we've detected that a watched key has been overwritten and we're watching it with this flag, a human readable error response like "value of key watched with WATCH_APPEND flag was overwritten" might be helpful.